rotary/sample.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+"""
+Created by: Lee Bergstrand (2023)
+
+Description: Classes and methods for representing Rotary sampling files.
+"""
+import os
+import re
+
+short_read_r_regex = re.compile("[Rr][12]")
+
+
+class SequencingFile(object):
+    """
+    An object representing a FASTQ file.
+    """
+
+    def __init__(self, file_path):
+        self.path = file_path
+        self.name = os.path.basename(self.path)
+
+        if 'fastq' not in self.name:
+            raise ValueError('{} is not a fastq file.'.format(self.name))
+
+        if '_' in self.name:
+            identifier, remaining = self.name.split('_', 1)  # Split on first _ if found.
```

Comment:
I can think of cases where keeping the text before the first underscore could lead to naming collisions between different samples (e.g., if someone had samples like `P_putida`, `P_aeruginosa`, etc.). Might it be better to strip off the final R1 / R2 and then keep everything before that as the sample ID? Not sure...

Reply:
Moved to issue